projects/storefrontlib/src/cms-components/cart/cart-shared/cart-item-list/cart-item-list.component.ts

```diff
@@ -78,12 +79,23 @@ export class CartItemListComponent {
         return entry;
       });
     } else {
-      this._items = items;
+      if (this.featureConfigService?.isLevel('3.1')) {
+        for (let i = 0; i < items.length; i++) {
+          if (JSON.stringify(this._items?.[i]) !== JSON.stringify(items[i])) {
+            this._items[i] = items[i];
+          }
+        }
```

Comment:
Although I understand it, the reason of adding this logic is not obvious. Can we add a code comment? For example:
```ts
// We'd like to avoid the unnecessary re-renders of unchanged cart items after the data reload.
// OCC cart entries don't have any unique identifier that we could use in Angular `trackBy`.
// So we update each array element to the new object only when it's any different to the previous one.
```